File: 175_invert-binary-tree/invert-binary-tree.java
/*
@Copyright:LintCode
@Author:   shaman
@Problem:  http://www.lintcode.com/problem/invert-binary-tree
@Language: Java
@Datetime: 16-03-27 15:02
*/

/**
 * Definition of TreeNode:
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left, right;
 *     public TreeNode(int val) {
 *         this.val = val;
 *         this.left = this.right = null;
 *     }
 * }
 */
public class Solution {
    public void invertBinaryTree(final TreeNode root) {
        if (root == null) return;
        if (root.left != null) invertBinaryTree(root.left);
        if (root.right != null) invertBinaryTree(root.right);
        TreeNode temp = root.left;
        root.left = root.right;
        root.right = temp;
    }
}

File: 175_invert-binary-tree/invert-binary-tree.py
# coding:utf-8
'''
@Copyright:LintCode
@Author:   shaman
@Problem:  http://www.lintcode.com/problem/invert-binary-tree
@Language: Python
@Datetime: 16-03-05 09:55
'''

"""
Definition of TreeNode:
class TreeNode:
    def __init__(self, val):
        self.val = val
        self.left, self.right = None, None
"""
class Solution:
    # @param root: a TreeNode, the root of the binary tree
    # @return: nothing
    def invertBinaryTree(self, root):
            if root is None:
                return None
            if root.left:
                self.invertBinaryTree(root.left)
            if root.right:
                self.invertBinaryTree(root.right)
            root.left, root.right = root.right, root.left
            return root